File: ts/src/entity/FlashEntity.ts

import { inspect } from 'node:util'

import { ThesmsworksEntityBase } from '../ThesmsworksEntityBase'

import type {
  ThesmsworksSDK,
} from '../ThesmsworksSDK'


import type {
  Operation,
  Context,
  Control,
} from '../types'

import type {
  Flash,
} from '../ThesmsworksTypes'

// TODO: needs Entity superclass
class FlashEntity extends ThesmsworksEntityBase<Flash> {

  constructor(client: ThesmsworksSDK, entopts: any) {
    super(client, entopts)
    this.name = 'flash'
    this.name_ = 'flash'
    this.Name = 'Flash'
  }


  make(this: FlashEntity) {
    return new FlashEntity(this._client, this.entopts())
  }







}


export {
  FlashEntity
}

File: dart/lib/entity/FlashEntity.dart

// ignore_for_file: non_constant_identifier_names

import 'dart:async';

import '../ThesmsworksEntityBase.dart';
import '../utility/ErrUtility.dart';

// Typed models: see ../ThesmsworksTypes.dart (Flash and the per-op request/match types).
class FlashEntity extends ThesmsworksEntityBase {
  FlashEntity(dynamic client, dynamic entopts) : super(client, entopts) {
    name = 'flash';
    name_ = 'flash';
    Name = 'Flash';
  }

  FlashEntity make() {
    return FlashEntity(client, entopts());
  }





}
